src/ParameterizedFunctionBase.cpp

```diff
@@ -208,48 +208,54 @@ void ParameterizedFunctionBase<MemorySpace>::SetCoeffs(Kokkos::View<const double
     SetCoeffsInternal(this->numCoeffs, this->savedCoeffs, coeffs);
 }
 
-template<typename MemorySpace>
-void ParameterizedFunctionBase<MemorySpace>::WrapCoeffs(Kokkos::View<double*, Kokkos::HostSpace> coeffs){
+template<typename MemorySpaceSrc, typename MemorySpaceDest>
+void WrapCoeffsInternal(unsigned int numCoeffs, 
+	Kokkos::View<double*,MemorySpaceDest>& coeffsDest,
+        Kokkos::View<double*,MemorySpaceSrc> coeffsSrc){
 
-    if(coeffs.size() != numCoeffs){
+    if(coeffsSrc.size() != numCoeffs){
         std::stringstream msg;
-        msg << "Error in ParameterizedFunctionBase<MemorySpace>::WrapCoeffs.  Expected coefficient vector with size " << numCoeffs << ", but new coefficients have size " << coeffs.size() << ".";
+        msg << "Error in ParameterizedFunctionBase<MemorySpace>::WrapCoeffs.  Expected coefficient vector with size " << numCoeffs << ", but new coefficients have size " << numCoeffs << ".";
         throw std::invalid_argument(msg.str());
     }
-    this->savedCoeffs = coeffs;
+    Kokkos::deep_copy(coeffsDest, coeffsSrc);
```

Comment:
This is going to be a little more complicated: WrapCoeffs should basically only make a shallow copy of the coefficients you give me. An analogy would be to imagine the following C syntax
```
void foo(double *x, double *y) {
    x = y; // WrapCoeffs
    *x = *y; // SetCoeffs
}
```

Here, you should only be setting the address of `coeffsDest` to `coeffsSrc`. In that same vein, `WrapCoeffs` should probably only work when `coeffsSrc` and `coeffsDest` are in the same memory space (this begs the question if an internal function like this is actually necessary, which I didn't consider before).